internal/app/udevd/main.go

```diff
@@ -10,50 +10,46 @@ import (
 	"fmt"
 	"log"
 	"os"
-	"path"
+	"path/filepath"
 
+	"github.com/mdlayher/kobject"
 	"github.com/pkg/errors"
 	"github.com/talos-systems/talos/internal/app/udevd/internal/drivers/scsi"
-	"github.com/talos-systems/talos/internal/app/udevd/internal/uevent"
 )
 
 func init() {
 	log.SetFlags(log.Lshortfile | log.Ldate | log.Lmicroseconds | log.Ltime)
 }
 
 func watch() error {
-	k, err := uevent.Dial()
+	client, err := kobject.New()
 	if err != nil {
-		return err
+		log.Fatal(err)
 	}
-	// nolint: errcheck
-	defer k.Close()
-
-	for u := range k.Watch() {
-		if u.Error != nil {
-			log.Printf("uevent error: %+v\n", err)
-			continue
+	for {
+		event, err := client.Receive()
+		if err != nil {
+			panic(err)
```

Comment:
I'm not sure panic is what you really want in these two calls.

Reply:
Ahh yes, I do not.